pkg/apis/pipeline/v1/matrix_types_test.go

```diff
@@ -67,114 +67,621 @@ func TestMatrix_FanOut_ToParams(t *testing.T) {
+			}, {
+				"GOARCH": "I-do-not-exist",
+			}},
+			expectedParams: []Params{{
+				{
+					Name:  "GOARCH",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "linux/amd64"},
+				}, {
+					Name:  "context",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/go117/context"},
+				}, {
+					Name:  "package",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"},
+				}, {
+					Name:  "version",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "go1.17"},
+				},
+			}, {
+				{
+					Name:  "GOARCH",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "linux/ppc64le"},
+				}, {
+					Name:  "context",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/go117/context"},
+				}, {
+					Name:  "package",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"},
+				}, {
+					Name:  "version",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "go1.17"},
+				},
+			}, {
+				{
+					Name:  "GOARCH",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "linux/s390x"},
+				}, {
+					Name:  "context",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/go117/context"},
+				}, {
+					Name:  "flags",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "-cover -v"},
+				}, {
+					Name:  "package",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"},
+				}, {
+					Name:  "version",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "go1.17"},
+				},
+			}, {
+				{
+					Name:  "GOARCH",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "linux/amd64"},
+				}, {
+					Name:  "package",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"}}, {
+					Name:  "version",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "go1.18.1"},
+				},
+			}, {
+				{
+					Name:  "GOARCH",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "linux/ppc64le"},
+				}, {
+					Name:  "package",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"},
+				}, {
+					Name:  "version",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "go1.18.1"},
+				},
+			}, {
+
+				{
+					Name:  "GOARCH",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "linux/s390x"},
+				}, {
+					Name:  "flags",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "-cover -v"},
+				}, {
+					Name:  "package",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"},
+				}, {
+					Name:  "version",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "go1.18.1"},
+				},
+			}, {
+				{
+					Name:  "GOARCH",
+					Value: ParamValue{Type: ParamTypeString, StringVal: "I-do-not-exist"},
+				},
+			}},
+		}}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			tt.matrix.FanOut()
```

Comment:
This line can be removed--should have caught it in an earlier PR.

Reply:
@lbernick Why should this be removed?  Shouldn't it be adding a new combination?

Reply:
I don't think `tt.matrix.FanOut()` mutates the input matrix; it just returns a list of params if I understand correctly. The next line calls `tt.matrix.FanOut` again, and uses the output it generates.

Reply:
@lbernick Not sure I follow.. Is this comment related to this removing this value: 
`Value: ParamValue{Type: ParamTypeString, StringVal: "I-do-not-exist"},`

Reply:
No, it's related to removing `tt.matrix.FanOut()`. Could this test case be simplified to:
```
for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if d := cmp.Diff(tt.want, tt.matrix.FanOut()); d != "" {
				t.Errorf("Combinations of Parameters did not match the expected Params: %s", diff.PrintWantGot(d))
			}
``` 
?

Reply:
I think this can be simplified without failing the tests, but would like to confirm with @pritidesai since she refactored this is in the previous PR: https://github.com/tektoncd/pipeline/pull/6312

cc: @Jerop

Reply:
I agree with @lbernick that it should be removed, I believe it was just missed in a previous review

Reply:
Okay, removed the `expectedParams` and simplified the test